ecs/alb-service/main.tf

```diff
@@ -0,0 +1,208 @@
+  description = "Comma separated list of subnet IDs used by the external ELB"
+}
+
+variable "internal_subnet_ids" {
+  description = "Comma separated list of subnet IDs used by the internal ELB"
+}
+
+variable "vpc_id" {
+    type = "string"
+}
+
+variable "security_groups" {
+  description = "Comma separated list of security group IDs for the external ELB"
+}
+
+variable "internal_security_groups" {
+  description = "Comma separated list of security group IDs for the internal ELB"
+}
+
+variable "port" {
+  description = "The container host port"
+}
+
+variable "cluster" {
+  description = "The cluster name or ARN"
+}
+
+variable "log_bucket" {
+  description = "The S3 bucket ID to use for the ELB"
+}
+
+variable "ssl_certificate_id" {
+  description = "SSL Certificate ID to use"
+}
+
+variable "iam_role" {
+  description = "IAM Role ARN to use"
+}
+
+variable "external_dns_name" {
+  description = "The subdomain under which the ELB is exposed externally, defaults to the task name"
+  default     = ""
+}
+
+variable "internal_dns_name" {
+  description = "The subdomain under which the ELB is exposed internally, defaults to the task name"
+  default     = ""
+}
+
+variable "external_zone_id" {
+  description = "The zone ID to create the record in"
+}
+
+variable "internal_zone_id" {
+  description = "The zone ID to create the record in"
+}
+
+/**
+ * Options.
+ */
+
+variable "healthcheck" {
+  description = "Path to a healthcheck endpoint"
+  default     = "/"
+}
+
+variable "healthcheck_interval" {
+  default = 30
+}
+
+variable "healthcheck_timeout" {
+  default = 5
+}
+
+variable "container_port" {
+  description = "The container port"
+  default     = 3000
+}
+
+variable "desired_count" {
+  description = "The desired count"
+  default     = 2
+}
+variable "deployment_maximum_percent" {
+  description = "The maximum capacity increase during deployment"
+  default = 200
+}
+
+variable "deployment_minimum_healthy_percent" {
+  description = "The minimum capacity during deployment"
+  default = 100
+}
+
+variable "stack_name" {
+  description = "stack name to use as the value for the Terraform tag"
+  default     = ""
+}
+
+variable "task_name" {
+  description = "ecs task name"
+}
+
+variable "task_definition" {
+  description = "ecs task definition"
+}
+
+variable "elb_idle_timeout" {
+  default = 30
+}
+
+resource "aws_ecs_service" "main" {
+  name            = "${var.task_name}"
+  cluster         = "${var.cluster}"
+  task_definition = "${var.task_definition}"
+  desired_count   = "${var.desired_count}"
+  iam_role        = "${var.iam_role}"
+
+  load_balancer {
+    target_group_arn ="${module.elb.external_target_id}"
+    container_name = "${var.task_name}"
+    container_port = "${var.container_port}"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  deployment_maximum_percent = "${var.deployment_maximum_percent}"
+  deployment_minimum_healthy_percent = "${var.deployment_minimum_healthy_percent}"
+}
+
+
+module "elb" {
```

Comment:
it's an elb, just new subset. doing some of this caused minimal amount of change